web/src/scheduled_messages_feed_ui.ts

```diff
@@ -33,13 +35,17 @@ function get_scheduled_messages_matching_narrow() {
                 return true;
             }
         } else if (scheduled_message.type === "stream") {
-            if (narrow_state.stream_sub() === undefined) {
+            if (narrow_state.stream_sub() === undefined || narrow_state.topic() === undefined) {
                 return false;
             }
             const narrow_dict = {
-                stream_id: narrow_state.stream_sub().stream_id,
-                topic: narrow_state.topic(),
+                stream_id: narrow_state.stream_sub()!.stream_id,
+                topic: narrow_state.topic()!,
             };
+            assert(
+                typeof scheduled_message.to === "number",
+                "stream_id has to be a unique number.",
+            );
```

Comment:
You can avoid this assertion by changing the `ScheduledMessage` type definition to be more precise as as a union of two types, where either `type="private"` and `to` is an array, or `type="stream"` and it's a number.

```
type ScheduledMessage = {                                                                               
    scheduled_message_id: number;                                                                       
    type: string;                                                                                       
    to: number[] | number;                                                                              
    topic: string;                                                                                      
    content: string;                                                                                    
    rendered_content: string;                                                                           
    scheduled_delivery_timestamp: number;                                                               
    failed: boolean;                                                                                    
};                                                                                                      
```

I'd do that refactor in a preparatory commit.

Reply:
Yes...that would be great......done!